core/rlwe/ciphertext.go

```diff
@@ -43,6 +43,18 @@ func NewCiphertextRandom(prng sampling.PRNG, params ParameterProvider, degree, l
 	return
 }
 
+// Plaintext casts the target ciphertext into a plaintext type.
+// This method is allocation free.
+func (ct Ciphertext) Plaintext() *Plaintext {
```

Comment:
This doesn't change the modulus, only re-interprets the ciphertext (degree 1 poly in s) as a degree 0 poly in s. It can for example be used after key-switching a ciphertext to the key zero in a multi-party setting (instead of having to allocate a plaintext and manually setting it's internal value to turn first poly of the key-switches ciphertext, which would require a good knowledge of the low level elements library).

Reply:
Thanks for the information,  is the same method used for both CKKS and BGV/BFV? 
In CKKS, we agree that after encoding the plaintext and ciphertext space are the same. But that is not true for BFV and BGV.

Reply:
Actually, it should be seen is a helper method for the following case: assume you have a ciphertext $(c_0, c_1)$, and you evaluate a decryption in place that modifies it to $(c_0 + sc_1, c_1)$, then by using `.Plaintext()` you wrap $(c0 + sc_1)$ in an `rlwe.Plaintext`, allowing to pass it to the `[scheme].Encoder`. Another case is, given a ciphertext $(0, 0)$, you could cast it as a plaintext to give it to an encoder and return $(encode(m), 0)$, and then add a mask on top of it. There are plenty of little cases that this helper makes easier to code, and it also helps it code readability.

Actually, the plaintext space of CKKS is not the same as the ciphertext space. The plaintext space of CKKS is $\mathbb{R}[X]/(X^N+1)$, so unbounded, while the ciphertext space is $\mathbb{Z}_{Q}[X]/(X^N+1)$, so bounded by $Q$. This is why $|m(X)|$ should always be smaller than $Q$.

Regarding the plaintext space of BGV/BFV, that depends how it is looked at: is it scaled by $t^{-1}\mod Q$ -> BFV or not -> BGV. Switching between both representations is trivial. By default, the encoder will scale plaintext by $t^{-1}\mod Q$ so it will assume that ciphertext encrypt such plaintext when decoding,